src/test/java/com/microsoft/graph/functional/OneNoteTests.java

```diff
@@ -429,6 +434,59 @@ public void testMultipartPost(){
             fail("Unable to write to output stream");
         }
     }
+    
+    /**
+     * Test posting multipart content to a page
+     */
+    @Test
+    public void testMultipartPostWithHeadersMap(){
```

Comment:
This is useful but what I'd like to see is coverage of the `createPartHeader` method which has a few conditional branches. If you wrote those simple tests you would see the formatting problems yourself and they would not need review iterations. Because you didn't explicilty add a test for `createPartHeader` I'm going to run through what I would expect to see for testing of a method like it:

* make `createPartHeader` static (including whatever state was used in its parameters)
* give the method default visibility
* add the annotation @VisibleForTesting or add comment // VisibleForTesting
* write numerous small test methods that cover every branch and condition of `createPartHeader` only (15 minutes of work to knock up)
* Keep the unit tests low level, only testing `createPartHeader` 

@darrelmiller I know we have discussed briefly how much unit testing to do but this is an obvious easy win and easy to write. I'm happy for testing paths not to be coverered but it should be deliberate and argued as to why it is not being done. 

On closer inspection the current test method performs no useful assertions so gives us no coverage at all of the changes made!

Reply:
- Wrote unit tests for createPartHeader. 
- createPartHeader cannot be made static because it uses addBoundary function which uses Boundary string that can be different for different multiparts. Suppose, developer wants to use Multipart at multiple places in application at those places the boundary value will be different so it cannot be made static, in turn then addBoundary and createPartHeader cannot be made static because of usage of non-static boundary member.